src/tabular-api-surface.ts

```diff
@@ -507,7 +520,13 @@ Please use the format 'prezzy' or '${instance.name}/tables/prezzy'.`);
           numRequestsMade++;
           if (
             numConsecutiveErrors <= maxRetries &&
-            (RETRYABLE_STATUS_CODES.has(error.code) || isRstStreamError(error))
+            (RETRYABLE_STATUS_CODES.has(error.code) ||
+              isRstStreamError(error)) &&
+            !(
+              timeout &&
+              error.code === grpc.status.DEADLINE_EXCEEDED &&
```

Comment:
The error code doesn't matter here. Client should fail with deadline exceeded error after the total timeout expires.

Reply:
Sounds good. Now that it is understood that it is a requirement that the timeout should be respected for all error codes so I've removed this line accordingly.